common/djangoapps/xblock_django/management/commands/ensure_indexes.py

```diff
@@ -0,0 +1,60 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.conf import settings
+import importlib
+
+from xmodule.contentstore.mongo import MongoContentStore
+# from xmodule.modulestore.mongo import MongoModuleStore
+# from xmodule.modulestore.split_mongo.split_draft import DraftVersioningModuleStore
+
+class Command(BaseCommand):
+
+    args = ''
+    help = 'Creates the indexes for ContentStore and ModuleStore mongodb databases'
+
+    def handle(self, *args, **options):
+        # Create the indexes for the Contentstore
+        grid_fs_instance = self.create_grid_fs_content_store(settings.CONTENTSTORE)
+        self.ensure_indexes(grid_fs_instance)
+
+        # Create the indexes for the Modulestore. The creation is intended to be data driven
+        # via the items in settings in MODULESTORE['default']['OPTIONS']['stores']
+        for store in settings.MODULESTORE['default']['OPTIONS']['stores']:
+            # Create an instance of the class using the supplied engine spcified in the configuration.
+            module_name = store['ENGINE'][:store['ENGINE'].rfind('.')]
+            class_name = store['ENGINE'][store['ENGINE'].rfind('.') + 1:]
+            # Now get a reference to the class
+            module = importlib.import_module(module_name)
+            class_ = getattr(module, class_name)
+            mongo_instance = self.create_mongo_module_store(store, class_)
+            self.ensure_indexes(mongo_instance)
+
+    def create_grid_fs_content_store(self, settings):
+        '''
+        :param settings: a dict containing the keys/values for CONTENTSTORE
+        :return: MongoContentStore instance
+        '''
+        host = settings['DOC_STORE_CONFIG'].get('host', 'localhost')
+        port = settings['DOC_STORE_CONFIG'].get('port', 27017)
+        db = settings['DOC_STORE_CONFIG']['db']
+        return MongoContentStore(host, db, port, None, None)
+
+    def create_mongo_module_store(self, settings, class_):
+        '''
+        :param settings: a dict containing the keys/values for MODULESTORE
+        :param class_: ABCMeta
+        :return: MongoContentStore ?
+        '''
+        fs_root = settings['OPTIONS'].get('fs_root', '.')
+        return class_(contentstore=None,
+                      doc_store_config=settings['DOC_STORE_CONFIG'],
+                      fs_root=fs_root,
+                      render_template=None
+                      )
+
+    def ensure_indexes(self, content_store):
+        '''
+        Calls ensure_index on the modulestore passed in.
+        :param content_store:
+        :return:
+        '''
+        content_store.ensure_indexes()
```

Comment:
The naming is a bit odd here. The docstring refers to it as a modulestore, and the variable is a `content_store`.